src/mining/collapse.ts

```diff
@@ -0,0 +1,135 @@
+import { Float } from 'java.lang';
+import { Location, Material, Sound, SoundCategory } from 'org.bukkit';
+import { Block } from 'org.bukkit.block';
+import { Player } from 'org.bukkit.entity';
+import { BlockBreakEvent } from 'org.bukkit.event.block';
+import { Vector } from 'org.bukkit.util';
+
+const COLLAPSE_CHECK_CHANCE = 1;
+
+const STONES = [
+  Material.STONE,
+  Material.DIORITE,
+  Material.GRANITE,
+  Material.ANDESITE,
+];
+
+const COLLAPSE_RADIUS = 10;
+const SUPPORT_RADIUS = 6;
+const SUPPORTS = [
```

Comment:
If possible, use ES6 Set.

Reply:
Changed to Set. Had to use ordinal or string for now